### ISSUE TITLE
convert_points_from_homogeneous divides by w + eps instead of w: 33% error at small positive w, 100% at small negative w

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#3938]
`convert_points_from_homogeneous` divides by `z_vec + eps` rather than by `z_vec` (`kornia/geometry/conversions.py:200-203`):

```python
mask: torch.Tensor = torch.abs(z_vec) > eps
scale = torch.where(mask, 1.0 / (z_vec + eps), torch.ones_like(z_vec))
```

The guard `torch.where(|w| > eps, ...)` already handles the near-zero case, so adding `eps` to the denominator serves no protective purpose — it only biases the result. Because `eps` is added without regard to `sign(w)`, it *inflates* a small positive denominator and *shrinks* a small negative one, so the error is asymmetric in the sign of `w` and can reach 33 % / 100 % just past the threshold. Away from the threshold the result is still never exactly `numerator / w`; it carries a permanent relative bias of `-eps/w`.

The code comment on the lines above points at [opencv/opencv#14411](https://github.com/opencv/opencv/pull/14411/files) as the intended reference, and OpenCV's `convertPointsFromHomogeneous` uses `scale = (fabs(w) > FLT_EPSILON) ? 1./w : 1.` — i.e. plain `1/w`, with no epsilon in the denominator.

## Repro

```python
import torch
from kornia.geometry.conversions import convert_points_from_homogeneous, convert_points_to_homogeneous

f64 = torch.float64
for w in [2e-8, -2e-8, 2.0, -2.0]:
    p = torch.tensor([[2., 4., w]], dtype=f64)
    out = convert_points_from_homogeneous(p)
    print(f"w={w!r:>8} -> {out.tolist()}   exact={[2.0 / w, 4.0 / w]}")

rt = convert_points_from_homogeneous(convert_points_to_homogeneous(torch.tensor([[1.5, -2.5]], dtype=f64)))
print("to->from round-trip of [[1.5, -2.5]]:", rt.tolist())
```

## Observed

```
w=   2e-08 -> [[66666666.66666666, 133333333.33333331]]   exact=[100000000.0, 200000000.0]
w=  -2e-08 -> [[-200000000.0, -400000000.0]]   exact=[-100000000.0, -200000000.0]
w=     2.0 -> [[0.999999995, 1.99999999]]   exact=[1.0, 2.0]
w=    -2.0 -> [[-1.000000005, -2.00000001]]   exact=[-1.0, -2.0]
to->from round-trip of [[1.5, -2.5]]: [[1.4999999850000003, -2.4999999750000006]]
```

## Expected

- `w = +2e-8` should give `[1e8, 2e8]`. Observed `[6.667e7, 1.333e8]` is **33 % low**, because `1/(2e-8 + 1e-8) = 3.33e7` instead of `5e7`.
- `w = -2e-8` should give `[-1e8, -2e8]`. Observed `[-2e8, -4e8]` is **100 % high** — the epsilon shrinks `|w|` from `2e-8` to `1e-8` instead of growing it. Two inputs that differ only in the sign of `w` therefore get errors in opposite directions and of different magnitudes, which no consistent regularisation should do.
- `w = 2.0` in float64 should give exactly `[1.0, 2.0]`. Observed `[0.999999995, 1.99999999]` is off by `5e-9` relative even though nothing here is near-degenerate; the `eps` bias applies to every point, always.
- `convert_points_from_homogeneous(convert_points_to_homogeneous(p))` should be an identity to machine precision. Observed it is an identity only to ~1e-8 relative, which is *worse than float32* despite the inputs being float64.

Suggested fix — divide by `w` on the masked branch, using the standard double-`where` so the backward pass stays NaN-free (cf. the NaN-gradient report in #367). Only the `+ eps` in the denominator is at fault; the `|w| > eps` passthrough behavior should be preserved as-is:

```python
mask = torch.abs(z_vec) > eps
safe_z = torch.where(mask, z_vec, torch.ones_like(z_vec))
scale = torch.where(mask, 1.0 / safe_z, torch.ones_like(z_vec))
```

Executed with that substitution, on the same inputs as above:

```
w= 2e-08 -> [[100000000.0, 200000000.0]]     (exact)
w=-2e-08 -> [[-100000000.0, -200000000.0]]   (exact)
w=   2.0 -> [[1.0, 2.0]]                     (exact)
w=  -2.0 -> [[-1.0, -2.0]]                   (exact)
w=   0.0 -> [[2.0, 4.0]]                     (passthrough preserved)
w= 1e-12 -> [[2.0, 4.0]]                     (passthrough preserved)
grad of sum() w.r.t. [[2., 4., 0.]] -> [[1.0, 1.0, 0.0]]   (finite, no NaN)
```

## Classification

**Clearly-broken-output bug**, not a semantic/default-convention change. The documented contract is Euclidean division of a homogeneous point; returning `numerator / (w + eps)` is a wrong evaluation of that contract, not an alternative convention. Per the reference-core lifecycle (Part I), this class may land early as an ordinary bugfix rather than waiting for the coordinated Repair & Deprecation window. Caveat for the fix PR: every output changes by ~`eps/w` relative, so byte-identity on the default path cannot be preserved — the change is strictly toward the exact value, and downstream tolerances at or below 1e-8 relative may need review.

## Environment

kornia 0.9.0rc1 @ `3358ef16`, torch 2.9.1, cpu.

Posted on behalf of @ducha-aiki by Claude (Fable 5).

**Comment by ducha-aiki** [comment]
Closed by #3976, merged as `816e2c69` — recording the verification, since the auto-close left no note.

The `eps` no longer enters the division; it only selects the branch, so for `|w| > eps` the point is divided by exactly `w`. Verified on `main` at `816e2c69`:

```python
>>> import torch, kornia
>>> kornia.geometry.conversions.convert_points_from_homogeneous(torch.tensor([[2., 4., 2e-8]]))
tensor([[1.0000e+08, 2.0000e+08]])     # was [[6.6666668e7, 1.3333334e8]], 33 % low
```

The negative-`w` half is fixed with it: `w + eps` was not sign-aware, so it grew a small positive denominator and shrank a small negative one, giving errors in opposite directions for inputs differing only in the sign of `w`. `test_division_is_exact_for_both_signs_of_w` pins both at `atol = rtol = 0` (using `w = 2 ** -26`, so the quotients are powers of two and exact in every dtype that can hold them), and `test_roundtrip_from_to_homogeneous_is_identity` pins the oracle-free invariant that `from(to(x)) == x` bit for bit.

One behaviour the fix does *not* change, now stated in the Convention block instead of a warning: `eps` is compared against `w` in the working dtype, so in `float16` both the default `eps` and `w = 2e-8` underflow to `0` and the point takes the pass-through branch. That is pinned by `test_wart_float16_underflowed_default_eps_flips_branches`.

*Posted on behalf of @ducha-aiki by Claude (Opus 5).*
